### PR TITLE
Gives Public Mining on Icebox an Elevator, and just a TON of lift fixes so they're viable to map with. [MULTI-BEE] [:pslime: in the chat]

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

image
image

## Why It's Good For The Game

It's insanely hard to get mechas to go down zlevels. Like, seriously. Can't go down ladders with those fuckers. How about an elevator? Also should massively help with getting simplemobs, crates, and other whathaveyou back up to the station.

One thing I purposefully did to make the elevator a bit more interesting: It can't be called down, only called back up. So, if there isn't an elevator on the lower level you can't use it.

## Changelog
:cl:
add: Icebox now has an elevator! I just took the trousers of every shaft miner to batter town, word.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
can you emag it to send you to hell (lavaland)

**Comment by Barhandar** [comment]
Is it possible to build your own elevator or at least link up the button to it so bored engies can amend lack of one on the lower level?

**Comment by tralezab** [comment]
It's basically the same deal as the cremator, so if you can for that, you could for this... I think!

**Comment by tralezab** [comment]
Update: Lights are redone as to not be shitty and weird, being under an elevator when it lands will gib you

**Comment by lordpidey** [comment]
Out of sheer curiosity...

Will a supermatter shard get crushed by a falling elevator?

**Comment by tralezab** [comment]
Just living things!

**Comment by tralezab** [comment]
And yes, while I think gibbing is crazy lethal, it can only happen in elevator shafts, which have absolutely nothing and really just aren't places anyone should be

**Comment by lordpidey** [comment]
Add an interior door on the lower level.

And no safety checks.  Make there be a nice shortcut that just occasionally gibs you with no warning.  Really plays into the "Budget cut driven deathtrap" feeling of NT.

**Comment by tralezab** [comment]
@lordpidey the elevator is pretty good for a lot of things, so I wanted to kinda put it out of the way so it's a not... SO much better than the ladder

**Comment by tralezab** [comment]
Okay, made them more stable

File: code/game/objects/structures/industrial_lift.dm [hunk 1]
```diff
@@ -1,3 +1,8 @@
+
+//Booleans in arguments are confusing, so I made them defines.
+#define LOCKED 1
+#define UNLOCKED 0
```
Comment (Tlaltecuhtli) [inline code/game/objects/structures/industrial_lift.dm:4]: #undefine this at the end

File: code/game/objects/structures/industrial_lift.dm [hunk 2]
```diff
@@ -89,15 +109,28 @@
...
+/**
+  * Sets all lift parts's controls_locked variable. Used to prevent moving mid movement, or cooldowns.
+  */
+/datum/lift_master/proc/set_controls(state)
+	for(var/l in lift_platforms)
+		var/obj/structure/industrial_lift/lift_platform = l
+		lift_platform.controls_locked = state
+
+GLOBAL_LIST_EMPTY(lifts)
 /obj/structure/industrial_lift
```
Comment (Tlaltecuhtli) [inline code/game/objects/structures/industrial_lift.dm:134]: empty lines mess

File: code/modules/assembly/doorcontrol.dm [hunk 3]
```diff
@@ -146,3 +146,38 @@
 			C.cremate(usr)
 
 	addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 50)
+
+//purposefully keeping this deciseconds because we divide it
+#define FLOOR_TRAVEL_TIME 2 SECONDS
```
Comment (Tlaltecuhtli) [inline code/modules/assembly/doorcontrol.dm:151]: undefine this at the end

**Comment by tralezab** [comment]
2/3s of tlal's review made no sense or actively breaks my pr

File: code/game/objects/structures/industrial_lift.dm [hunk 4]
```diff
@@ -229,11 +271,12 @@
 
 /obj/structure/industrial_lift/proc/show_fluff_message(going_up, mob/user)
 	if(going_up)
-		user.visible_message("<span class='notice'>[user] move lift up.</span>", "<span class='notice'>Lift move up.</span>")
+		user.visible_message("<span class='notice'>[user] moves the lift upwards.</span>", "<span class='notice'>You move the lift upwards.</span>")
 	else
-		user.visible_message("<span class='notice'>[user] move lift down.</span>", "<span class='notice'>Lift move down.</span>")
+		user.visible_message("<span class='notice'>[user] moves lift downwards.</span>", "<span class='notice'>You move the lift downwards.</span>")
```
Comment (willox) [inline code/game/objects/structures/industrial_lift.dm:276]:
```suggestion
		user.visible_message("<span class='notice'>[user] moves the lift downwards.</span>", "<span class='notice'>You move the lift downwards.</span>")
```

**Review by Timberpoes — APPROVED** [review]

**Comment by Timberpoes** [comment]
Been hanging around long enough. Let's get it merged, if it breaks anything we'll fix it in post.

**Pull request merged**